ci/test_common.sh

```diff
@@ -160,16 +160,40 @@ run_py_benchmark() {
 }
 
 ################################## Distributed #################################
+install_distributed_dev_mode() {
+  # Running Distributed tests which access its internals requires installing it in
+  # developer mode. This isn't a great solution but it's what we can currently do
+  # to run non-public API tests in CI.
+
+  rapids-logger "Install Distributed in developer mode"
+  git clone https://github.com/dask/distributed /tmp/distributed
+  pip install -e /tmp/distributed
+  # `pip install -e` removes files under `distributed` but not the directory, later
+  # causing failures to import modules.
+  PYTHON_ENV_PATH=${CONDA_PREFIX:-/pyenv}
+  rm -rf $(find ${PYTHON_ENV_PATH} -type d -iname "site-packages")/distributed
```

Comment:
I had to do this as explained in the comment above. I think this only happens if you happen to run something that `import distributed` first and creates pycache and stuff. I've always had this problem when developing locally and I do not have a better solution. Just pinging here in case you know of a cleaner way @vyasr .

Reply:
Sorry, I had issued a /merge here already. If you have any ideas I'm happy to address it in a follow-up PR.